kie-spring/src/main/java/org/kie/spring/KModuleBeanFactoryPostProcessor.java

```diff
@@ -271,47 +279,47 @@ public void setApplicationContext(ApplicationContext applicationContext) throws
     /**
      * This is a HACK for web applications deployed to EAP/WildFly which are using kie-spring.
      *
-     * The method tries to figure out the root URL based on the provided app context. It covers (at least) the
+     * The method tries to figure out the root URL based on the provided URL enumeration. It covers (at least) the
      * following two use cases:
      *   1) kie-spring deployed together (bundled) with Spring webapp (inside WEB-INF/lib)
      *   2) kie-spring deployed as EAP module + Spring webapp depending on that module
      *
      * First of all it tries to determine if it is running on EAP, based on EAP specific resource URL. If that is the
-     * case it looks for the "WEB-INF/classes" dir and return that as an VFS URL (vfs:/...). Later on this URL
+     * case it looks for the "WEB-INF/classes" dir and return that as an VFS URL (vfs:/...). Later on, this URL
      * needs to be translated to a real filesystem path. This is one by {@link ClasspathKieProject#fixURLFromKProjectPath(URL)}
      *
-     * @param applicationContext Spring app context used by the application
+     * @param rootUrls Classpath root URLs
      * @return NULL is case the code is not running on EAP, otherwise root URL of the webapp context (that is webapp's WEB-INF/classes)
-     * @throws IOException in case of I/O error when gathering resources from app context's classloader
      */
-    private URL tryGetRootUrlForEapContext(ApplicationContext applicationContext) throws IOException {
-        Enumeration<URL> rootUrls = applicationContext.getClassLoader().getResources("/");
-        if (containsEapSpecificUrl(rootUrls)) {
-            while (rootUrls.hasMoreElements()) {
-                URL url = rootUrls.nextElement();
-                if (url.toString().endsWith(WEB_INF_CLASSES_FOLDER)) {
-                    return url;
-                }
+    URL tryGetRootUrlForEapContext(Enumeration<URL> rootUrls) {
+        boolean containsEapSpecificUrl = false;
```

Comment:
@mariofusco do we need to check for the "EAP-specific" URL? It seems that we could just search for the WEB-INF/classes URL and use that (but I may be overlooking something).